nautilus_core/model/src/data/stubs.rs

```diff
@@ -337,6 +337,10 @@ pub fn stub_bar() -> Bar {
         instrument_id,
         spec: bar_spec,
         aggregation_source: AggregationSource::External,
+
+        source_instrument_id: None,
+        source_spec: None,
+        source_aggregation_source: None,
```

Comment:
`source_aggregation_source` seems like a bit of an awkward name, maybe just `underlying_spec` and `underlying_aggregation_source`? (thats a long field name too.)

Or, `inner_spec` and `inner_aggregation_source`.
Or, `sub_spec` and `sub_aggregation_source`. :thinking:

Reply:
in my latest refactor, I'm using an enum for BarType, like here (happy to change the word composite to anything else, it doesn't change the problem, so we can change this). I think I've made some good progress this afternoon, the only error remaining in rust is from fmt (doesn't mean we're out of the woods).

```rust
pub enum BarType {
    Bar {
        /// The bar types instrument ID.
        instrument_id: InstrumentId,
        /// The bar types specification.
        spec: BarSpecification,
        /// The bar types aggregation source.
        aggregation_source: AggregationSource,
    },
    CompositeBar {
        /// The bar types instrument ID.
        instrument_id: InstrumentId,
        /// The bar types specification.
        spec: BarSpecification,
        /// The bar types aggregation source.
        aggregation_source: AggregationSource,

        /// The source bar types instrument ID.
        composite_instrument_id: InstrumentId,
        /// The source bar types specification.
        composite_spec: BarSpecification,
        /// The source bar types aggregation source.
        composite_aggregation_source: AggregationSource,
    },
}
```

Reply:
I think the `Composite` naming convention is pretty good. The enum is an elegant solution.

I wasn't sure how Cython might handle this, then I see its using a tagged union. Luckily we don't have to get our hands too dirty with raw C code written "by hand".

Reply:
which command allows to generate model.h and model.pxd automatically ?

Reply:
It's happening "automatically" through `cargo` and `cbindgen`.